include/flatbuffers/flatbuffers.h

```diff
@@ -437,14 +437,14 @@ class DetachedBuffer {
                  size_t reserved, uint8_t *cur, size_t sz)
     : allocator_(allocator), own_allocator_(own_allocator), buf_(buf),
       reserved_(reserved), cur_(cur), size_(sz) {
-    assert(allocator_);
+    assert(allocator_ != nullptr);
```

Comment:
The style thru-out the FlatBuffers code is to check pointers without `!= nullptr`. So rather than adding it in a few more places, it would be more consistent to remove it everywhere.